### DISCUSSION TITLE
push_firmware 7.50 FIT-Boxen / ftp-format

### DISCUSSION THREAD

**Discussion by maverick3005** [Freetz-NG/freetz-ng#790] [push firmware] [answered]
@fda77 

Du wolltest doch vor ner Weile "Wireshark einer recovery 7.50" um das push_firmware für alle FIT-Boxen mit neuerer Firmware zu fixen:

Das hier ist im Grunde auch das, was ich mit peters fit-tools im Vergleich schon mal geposted hatte:

SETENV kernel_args_tmp "avm_fwupdate mtdram1=0x7d240000,0x80000000 mtdparts_ext=update-image.0:0x2dc0000@0x0(fit-image)"

Zuvor werden regulär wie gehabt das Enviroment ausgelesen und die Bank bestimmt.

**Comment by fda77** [comment]
Ich hab keine Ahnung um was es hier geht

**Comment by maverick3005** [comment]
...um ein von dir geclostes ticket oder eine diskussion das push_firmware mit den neuen kernel commandline args nicht mehr klar kommt.
(direktes parsing problem, was fit-tools vom peter aber problemlos parsen können...)
dazu gabs auch 2 dateianhänge von 7.30 + 7.50 zum vergleich das lediglich die kernel-commandline sich geändert hat.
Eben das kann das PF-script noch nicht.

Zitat dazu von Dir damals: "Mach nen Wiresharklog von einer recovery 7.50".

Die eine zeile ist der Unterschied wo das PF bei ALLEN fit > 7.3x versionen mit mtd-ram ja die bekannten probleme hat.

Wenn Dir die exakte zeile nicht ausreicht kann ich gerne noch weitere Infos aus dem logfile zur Verfügung stellen.
In dem damaligen Thread hatte ich noch von einer 7.30 die ftp-kommunikation geloggt gehabt, was Dir nicht ausreichend war.

**Comment by maverick3005** [comment]
da war das drin:

https://github.com/Freetz-NG/freetz-ng/issues/744

nun hast Du die fehlende Info die Du damals haben wolltest :-)

**Comment by fda77** [comment]
Dir ist schon klar dass es je Gerät andere Adressen sind können? Funktioiert die in Freetz hinterlegt nicht mit push firmware?

**Reply by maverick3005** [reply]
das hatten wir doch alles todiskutiert damals in dem closed ticket und Du dicken screenshot gemacht noch :-)

nimm einfach mal ein STOCK-Image einer beliebigen 7.50er fit box und versuchs zu pushen, dann siehst sofort nochmal das problem :-ö)

Gilt für ALLE fit-boxen. JETZT ist aber ein recovery 7.50 da zum abucken.

Der parser für die "neue kernel-commandline" müsste im Grunde nur im pf-script angepasst werden, dann würde das wieder laufen.
internes flashen mit AVM-webif und freetz-webif klappt ja nach wie vor problemlos. nur initial gleich ne 7.50 drauf geht eben nicht wegen der neuen kernel commandline seit Monaten direkt mit PF...

Das da kommt dann sofort bei Dir:

```
* Analyzing '5590_07.39.all-rev102864_labor_freetz-ng-21297-a1ddabda1_20230223-014300.image' ...
Error: it is not known how to flash a fit-boot image with 'rootfs_ram' method yet.
Error: run a recovery.exe and get a capture by wireguard or tcpdump to add it.
Error: meanwhile install an older firmware and flash this one by the web interface.
```

die antwort wie dann das FTP command aussieht habe ich dir als einzeiler aus dem Wireshark ja log jetzt geposted.

direkt danach geht das STOR der daten los. Es liegt also nur am Pre-Parser des PF wenn die neue Kernel-commandline in Verwendung ist und es einen rootfs_ram parameter da auf einmal mehr gibt :-)

**Comment by fda77** [comment]
Aaaaaaaaa sagt das doch gleich! Ich dache es ging um das spezifische Offset

**Reply by fda77** [reply]
Ich hatte vermutet dass es für das Flashen keine Rolle spielt, jetzt sieht man dass SETENV auch gleich geblieben ist: https://github.com/Freetz-NG/freetz-ng/blob/master/tools/push_firmware#L185
Daher sollte es reichen die Warnung zu entfernen. Aber teste zuerst mal eine 7.5 firmware: Wenn die "t is not known how to flash a fit-boot image with 'rootfs_ram' method yet." Meldung kommt drück ganz schnell "v" ...

**Reply by maverick3005** [reply]
der einfachste test ist eine stockfirmware > 7.31 von avm mit der habe ich es damals als erstes versucht. mal schaun was gleich passiert....

**Reply by Matze-S** [reply, marked as answer]
JUHUU, Danke @fda77, schade daß man das nicht im der Doku findet - ich war schon kurz vor'm aufgeben bzgl. fit-boot oder fit-image flash....

<html>
<body>
<legend style="margin-top: 5px; color: rgb(0, 0, 0); font: bold 14px sans-serif; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><br class="Apple-interchange-newline">FRITZ!Smart Gateway</legend>

Firmware: 265.07.52 rev104250 | Branding: avm
-- | --
Freetz: ng-5d1bddc03-MSC-ng23050-FSGW | Uptime: 1 min


</body>
</html>

**Reply by fda77** [reply]
@Matze-S Danke für die Rückmeldung, ich mache dann die Warnmeldung weg
Kannst du noch einen Screenshot von "AVM-Info" anhängen?

**Reply by Matze-S** [reply]
@fda77 Ich musste aber einige Patches/Workarounds anbringen, daß ich den Build durchbekommen habe - auch wegen meinem alten Ubuntu 18.04 LTS - unglaubliches war da festzustellen, was das fit-image korrumpiert hat.
Naja, das Image hatte ich heute morgen dann, seitdem suche ich nach einer Möglichkeit, das Ding drauf zu bekommen - es gibt ja so gut wie gar nix von AVM, auch kein Recovery, habe alles abgesucht. Deine 7.5er-Defaults mit dem schnellen 'v' tun jedenfalls prima :-))) FREU

![image](https://github.com/Freetz-NG/freetz-ng/assets/81502008/163586ca-dbc2-491b-baa9-5ea637f816e4)

**Reply by Matze-S** [reply]
PS: Vor dem FSGW habe ich auch endlich meinen 1200AX auf den aktuellen Stad gebracht (gab lange nix >7.30) - auch ein fit-image, funktioniert auch prima, war allerdings etwas einfacher, da 07.30 beim flashen bekannt....
danke, vielen dank jedenfalls für Deine Arbeit, @fda77, auch an @peterpawn, un alle anderen, die hieran mitgewirkt haben. :-D

**Reply by fda77** [reply]
Die Warnung kannst du vergessen, die ist nun weg: 29d2f4ae49d5dcb72d4b3ed5007e10c32e072df2 , und ff773989cdf9c6435227ed14df4ecb670601d1a4
Interessant dass das Gerät wirklich nur 2 Cores hat und 50 MB Ram für irgend was reserviert (nicht sichtbar) sind.

Wenn du probleme beim compilieren hattest, müsste das Linux die Ursache sein, ein Minimalimage lief problemlos durch:
```
STEP 3: PACK
  checking for left over version-control-system files
  integrate freetz info file into image
packing var.tar
creating filesystem image (SquashFS4-xz)
  SquashFS block size: 64 kB (65536 bytes)
copying kernel image
  kernel image size: 3.5 MB
copying filesystem image
  filesystem image size: 20.6 MB
packing fitimg file
  fitimg file size: 24.2 MB, max 50.0 MB, free 25.8 MB (27084216 bytes)
packing images/Gateway_07.52.all.image
  packed image file size: 24.5 MB (25661440 bytes)
source firmware: Gateway_en-de-es-fr-it-nl-pl 265.07.52 rev104250 {ALL} [MOVE21NL1] (07.03.2023 12:19:32)
  source image file size: 23.9 MB (25088000 bytes)
done.

FINISHED

real    2m45,854s
```

Für Benutzer ist es beim compilieren egal welches Format das Image am Ende hat, beim flashen mit push-firmware auch da es alle Formate kann. Ich weiss dass im DEB darüber berichtet wird dass man FIT weder bauen noch flashen kann nur über "geht nicht" gingen die Meldungen nie hinaus

**Reply by fda77** [reply]
Die 7.51 labor auf dem  1200 AX läuft mit Freetz?

**Reply by Matze-S** [reply]
![image](https://github.com/Freetz-NG/freetz-ng/assets/81502008/bf148777-33e8-4b6a-8a59-6158a4e0f827)

**Reply by fda77** [reply]
Danke. Vom 1200ax hatt ich schon einen Screenshot. Der hatte aber Bootloader v1.11714 und 181MB sichtbaren ram, deiner hat 180MB von 256 MB. Da müsste mal jemand nachschauen warum AVM das so macht

**Reply by Matze-S** [reply]
> Wenn du probleme beim compilieren hattest, müsste das Linux die Ursache sein, ein Minimalimage lief problemlos durch:

Nur eine Sache trat nicht bei einem Ubuntu 22 im WSL2 auf, die anderen würde ich nicht auf das Ubuntu schieben - gcc & co wurden selbstvständlich aufwändig hochgezogen.
Soll ich meine Auffälligkeiten hier mal irgendwie aufzählen?
Bei mir sind Workarounds in 4 kleinen Commits in meinem Fork eingefloseen, aber das ist natürlich sauber - die wahre Ursache/Hintergründe kenne ich (noch) nicht.

**Reply by fda77** [reply]
Beim 1200ax ist mir noch aufgefallen, dass kein "AVM-Standardpasswort" angezeigt wird. Hat der eins hinten drauf gedruckt? Oder kann Freetz es nur nicht finden?

Mach am besten ne neue Diskussion, hier das muss man immer aufklappen. Mein ältestes Ubuntu ist nur noch 20-lts
Oder verlink die commits doch einfach

**Reply by Matze-S** [reply]
> Beim 1200ax ist mir noch aufgefallen, dass kein "AVM-Standardpasswort" angezeigt wird. Hat der eins hinten drauf gedruckt? Oder kann Freetz es nur nicht finden?
> 
> Mach am besten ne neue Diskussion, hier das muss man immer aufklappen. Mein ältestes Ubuntu ist nur noch 20-lts Oder verlink die commits doch einfach

Der 1200ax war heute morgen fertig und hat schon ein Standard-Passwort bekommen, bei den build's hat man viel Zeit nebenher zum kofnigurieren.... :-)

Ich räume jetzt erstmal den ganzen kram bei mir auf, bringe noch die 6591 und den 2400 und die 7490 auf aktuelle Stände, und irgendwann die Tage mach ich das dann evlt., falls es sich bis dahin nicht erledigt - die meisten Punkte habe ich übrigens in ähnlicher Form beim Drüberfliegen hier auch irgendwo noch gefunden, evtl. muss ich erstmal schauen, ob da irgendwo noch offene Fragen oder sind, etc., ... aber nicht mehr heute. Dank für die Antworten :-D

**Reply by fda77** [reply]
Schon komisch dass AVM nur noch labor veröffentlicht
Ich meinte das  "AVM-Standardpasswort" mit dem man sich nach einem Werksresett beim AVM-Webif einloggt. Das haben nur die neueren Boxen, daher sollte die 1200ax es auch haben. Nur wurde das laut deinem Screenshot nicht gefunden/fehlt, beim Gateway aber schon

**Reply by Matze-S** [reply]
@fda77 Hi, bins doch nochmal,

ähem  evtl. gibt es ein Missverständnis; ich dachte, Du meinst die rote Meldung oben darüber, aber die bezieht sich doch nur auf den Admin oder root user, sobald ich die gesetzt hab, verschwindet die Meldung - glaube ich (mache das ja auch nur 1x pro Freetz Neuinstallation), jetzt sind jedenfalls beide Hinweise weg.

Das Passwort, daß Du meinst, wird doch nur im FritzOS als Default verwendet, bei freetz habe ich mich jedenfalls in beiden Fällen mit dem altbekannten freetz default pw eingeloggt, root musste ich beim ersten "login" (per SSH) vergeben, .... Hm

**Reply by fda77** [reply]
Genau, ich meinte das avm-pw. Auf dem Screenshot oben des Gateway wird es angezeigt (Doppelklick auf den Balken oder Mouse-Hoover).
Bei dem 1200ax dagegen nicht. Ich weiss jetzt aber nicht ob Freetz es nicht findet oder ob der 1200ax dies relativ neue Feature gar nicht hat. Da du so ein Gerät hast könntest du nachschauen ;-)

**Comment by maverick3005** [comment]
wenn Du auf Nummer sicher gehen willst das das Image in beiden Fällen bei fit valid ist anstelle die Warnung kompett rauszunehmen kannst du ja auch optional auf eine von beiden varianten noch "testen" bei den kernel args:

altes format/fit:

```
qcaarmv8_HW272_squashFS_filesystem {
            description = "squashFS for qcaarmv8_HW272";
            #address-cells = <0x00000001>;
            avm,data-align = <0x00001000>;
            data = /incbin/("image_003.bin"); // size: 28045312, offset=0x003400e0
            type = "filesystem";
            compression = "none";
            load = <0x43000000>;
            avm,kernel-args = "mtdram=ram-filesystem,0x43000000,0x44b00000 mtdparts_ext=ram-filesystem:28045312@0x0(rootfs_ram)";
            hash_0 {
                algo = "crc32";
                value = <0x5605953b>;
            };
        };
```

Neues Format > 7.31er versionen:

```
        qcaarmv8_HW272_squashFS_filesystem {
            description = "squashFS for qcaarmv8_HW272";
            avm,image-version = <0x00000001>;
            #address-cells = <0x00000001>;
            avm,data-align = <0x00001000>;
            data = /incbin/("image_003.bin"); // size: 31596544, offset=0x00355abc
            type = "filesystem";
            compression = "none";
            load = <0x43000000>;
            avm,kernel-args = "mtdram=rootfs_ram,0x43000000,0x44f00000";
            hash_0 {
                algo = "crc32";
                value = <0xc73c390e>;
            };
        };
```

**Reply by fda77** [reply]
Ausser den args wegen denen diese Warnung ausgegeben wird ist da nichts anders.
Wie du schriebst ist das fürs flashen wahrscheinlich egal.
Wenn du es getestet hast entfällt das "wahrscheinlich".
Wie du testen kannst schrieb ich ja schon

**Reply by maverick3005** [reply]
Ja eh, erst mal den check #remarken und einen dry run machen sobald ich das im Zugriff habe wieder.
